### PR TITLE
[15 min fix] Refactor private landing pages model code

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

This is a tiny refactor of the logic, to ensure data integrity at the model level.

Closes https://github.com/forem/forem/issues/13960

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: the point of this refactor is to have the same UX and tests but with a slightly different backend
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

File: app/models/page.rb [hunk 1]
```diff
@@ -9,11 +7,12 @@ class Page < ApplicationRecord
...
   validate :body_present
   validate :unique_slug_including_users_and_orgs, if: :slug_changed?
-  validate :single_landing_page, if: :will_save_change_to_landing_page?
 
   before_validation :set_default_template
   before_save :evaluate_markdown
-  after_save :bust_cache
+
+  after_commit :ensure_uniqueness_of_landinge_page
+  after_commit :bust_cache
```
Comment (rhymes) [inline app/models/page.rb:15]: This is not strictly related to this PR but we should get in the habit of enqueing async workers only when Rails has committed its transaction, see also https://github.com/forem/forem/pull/13969

File: app/controllers/admin/pages_controller.rb [hunk 2]
```diff
@@ -95,31 +101,5 @@ def prepopulate_new_form(slug)
                 Page.new
               end
     end
-
-    def update_and_overwrite_landing_page
```
Comment (rhymes) [inline app/controllers/admin/pages_controller.rb:99]: we don't need the whole concept of "overwrite landing page" anymore because the `Page` model will take care of this automatically

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -54,11 +54,9 @@ def verify_private_forem
     return if self.class.module_parent.to_s == "Admin"
     return if user_signed_in? || Settings::UserExperience.public
 
-    @page = Page.landing_page
-
     if api_action?
       authenticate!
-    elsif @page
+    elsif (@page = Page.landing_page)
```
Comment (rhymes) [inline app/controllers/application_controller.rb:59]: this was put here to avoid loading the landing page if the request is an API request...

**Review by fdoxyz — APPROVED** [review]

File: app/controllers/admin/pages_controller.rb (hunk 2 above)
Comment (juliannatetreault) [reply app/controllers/admin/pages_controller.rb:99]: It's so nice to see this gone!

**Review by juliannatetreault — APPROVED** [review]
This refactor is 😍! Thank you so much for taking the time to improve the code, @rhymes!
